### DISCUSSION TITLE
How to project shadow + Ambient Occlusion with a transparent/alpha zero object?

### DISCUSSION THREAD

**Discussion by Dereuz24** [bnpr/Malt#578] [Q&A]
I'm trying to isolate objects to render them in full view but with all shadows, lights and interactions still influencing them

An object with transparency and color with zero alpha does cast shadow

However the AO node doesn't. I tried to depth offset the object and the shadows still worked properly, however the AO disappeared just like it does for transparent objects. 

Is there any way to isolate AO in viewport, or in a render pass, or to have an object in isolation mode with everything (shadows, lights, AO, etc) but without any other object on screen?.

Edit: The only solution I came up with was to render without AO, then just render AO only of all objects and then mask each object individually all in post. But it would be convenient to be able to turn the opacity of an object to zero and still be able to influence other objects with AO. Can this be done?

EDIT2: The problem is that the other solution also requires me to create masks for objects overlapping the one I want to isolate and then turn their values all the way down to fill the area where these would be. That's a lot of layers to fix in post...

**Comment by pragma37** [comment]
Ambient Occlusion is a screen-space effect, so it can't evaluate anything that is not rendered.
There's no workaround other than using a different method for AO (like SDFs or raytracing), but implementing those in Malt would be far from trivial.

**Reply by Dereuz24** [reply]
Understood

I can get around some of those things with the compositor + eevee/cycles.

Are there limitations to scene linking while keeping malt nodes with regular materials at the same time? I've been combining them at small scale but haven't used it in large scenes in case I'm missing some compatibility issue
